tests/integration/golang/mlflow/namespace/flows/artifact_test.go

```diff
@@ -0,0 +1,363 @@
+		s.T(),
+		s.MlflowClient.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace2Code,
+		).WithQuery(
+			request.ListArtifactsRequest{
+				RunID: run1ID,
+			},
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			fmt.Sprintf("%s%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsListRoute),
+		),
+	)
+	assert.Equal(s.T(), fmt.Sprintf("RESOURCE_DOES_NOT_EXIST: unable to find run '%s'", run1ID), resp.Error())
+	assert.Equal(s.T(), api.ErrorCodeResourceDoesNotExist, string(resp.ErrorCode))
+
+	resp = api.ErrorResponse{}
+	assert.Nil(
+		s.T(),
+		s.MlflowClient.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithQuery(
+			request.ListArtifactsRequest{
+				RunID: run2ID,
+			},
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			fmt.Sprintf("%s%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsListRoute),
+		),
+	)
+	assert.Equal(s.T(), fmt.Sprintf("RESOURCE_DOES_NOT_EXIST: unable to find run '%s'", run2ID), resp.Error())
+	assert.Equal(s.T(), api.ErrorCodeResourceDoesNotExist, string(resp.ErrorCode))
+
+	// test `GET /artifacts/get` endpoint.
+	s.getRunArtifactAndCompare(namespace1Code, request.GetArtifactRequest{
+		RunID: run1ID,
+		Path:  "artifact1.file",
+	}, "content1")
+
+	s.getRunArtifactAndCompare(namespace2Code, request.GetArtifactRequest{
+		RunID: run2ID,
+		Path:  "artifact2.file",
+	}, "content2")
+
+	// test `GET /artifacts/get` endpoint.
+	// check that there is no intersection between runs, so when we request
+	// run 1 in scope of namespace 2 and run 2 in scope of namespace 1 API will throw an error.
+	resp = api.ErrorResponse{}
+	assert.Nil(
+		s.T(),
+		s.MlflowClient.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace2Code,
+		).WithQuery(
+			request.GetArtifactRequest{
+				RunID: run1ID,
+			},
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			fmt.Sprintf("%s%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsGetRoute),
+		),
+	)
+	assert.Equal(s.T(), fmt.Sprintf("RESOURCE_DOES_NOT_EXIST: unable to find run '%s'", run1ID), resp.Error())
+	assert.Equal(s.T(), api.ErrorCodeResourceDoesNotExist, string(resp.ErrorCode))
+
+	resp = api.ErrorResponse{}
+	assert.Nil(
+		s.T(),
+		s.MlflowClient.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithQuery(
+			request.ListArtifactsRequest{
+				RunID: run2ID,
+			},
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			fmt.Sprintf("%s%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsGetRoute),
+		),
+	)
+	assert.Equal(s.T(), fmt.Sprintf("RESOURCE_DOES_NOT_EXIST: unable to find run '%s'", run2ID), resp.Error())
+	assert.Equal(s.T(), api.ErrorCodeResourceDoesNotExist, string(resp.ErrorCode))
+}
+
+func (s *ArtifactFlowTestSuite) createRun(namespace string, req *request.CreateRunRequest) string {
```

Comment:
maybe this (and other common scenarios) should be attached to `BaseTestSuite` for reuse?

Reply:
In `BaseTestSuite` we usually store something that really needs probably in each test. I left this here for now because not fully sure where it should be. A agree that we have a bit of duplication right now, but I think we can improve it a bit later. In my mind I have an idea to have heigh level `client` for `mlflow` and `aim`, where client will have such constructions like that: `createRun`, `updateRun` and all the actual request building will be hidden under the hood. But again, this leads to a bit more changes.